packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -0,0 +1,204 @@
+    final vm = await serviceManager.service.getVM();
+    final futures = <Future>[];
+    for (final isolate in vm.isolates) {
+      futures.add(callback(isolate));
+    }
+    await Future.wait(futures);
+  }
+
+  Future<void> _setHttpTimelineRecording(bool state) async {
+    await _forEachIsolate((isolate) async {
+      final future = serviceManager.service
+          .setHttpEnableTimelineLogging(isolate.id, state);
+      // If the isolate is paused the request above will never complete.
+      await Future.any([
+        future,
+        Future.delayed(const Duration(milliseconds: 500)),
+      ]);
+    });
+    _httpRecordingNotifier.value = state;
+  }
+
+  /// Force refreshes the HTTP requests logged to the timeline.
+  Future<void> refreshRequests() async {
+    final timestamp = await serviceManager.service.getVMTimelineMicros();
+    final timeline = await serviceManager.service.getVMTimeline(
+        timeOriginMicros: _lastProfileRefreshMicros,
+        timeExtentMicros: timestamp.timestamp - _lastProfileRefreshMicros);
+    _lastProfileRefreshMicros = timestamp.timestamp;
+    _processHttpTimelineEvents(timeline);
+  }
+
+  /// Enables HTTP request recording on all isolates and starts polling.
+  /// 
+  /// If `alreadyRecording` is true, the last refresh time will be assumed to
+  /// be the beginning of the process (time 0).
+  Future<void> startRecording({
+    bool alreadyRecording = false,
+  }) async {
+    // Set the current timeline time as the time of the last refresh.
+    final timestamp = await serviceManager.service.getVMTimelineMicros();
+
+    if (!alreadyRecording) {
+      // Only include HTTP requests issued after the current time.
+      _lastProfileRefreshMicros = timestamp.timestamp;
+    }
+
+    // Determine the offset that we'll use to calculate the approximate
+    // wall-time a request was made. This won't be 100% accurate, but it should
+    // easily be within a second.
+    _timelineMicrosOffset =
+        DateTime.now().microsecondsSinceEpoch - timestamp.timestamp;
+
+    await resumeRecording();
+  }
+
+  /// Pauses the output of HTTP request information to the timeline.
+  /// 
+  /// May result in some incomplete timeline events.
+  Future<void> pauseRecording() async => await _setHttpTimelineRecording(false);
+
+  /// Resumes recording without resetting the last refresh timestamp.
+  Future<void> resumeRecording() async {
+    await _setHttpTimelineRecording(true);
+    _startPolling();
+  }
+
+  /// Checks to see if HTTP requests are currently being output. If so, recording
+  /// is automatically started upon initialization.
+  Future<void> initialize() async {
+    bool enabled = false;
+    await _forEachIsolate(
+      (isolate) async {
+        final future =
+            serviceManager.service.getHttpEnableTimelineLogging(isolate.id);
+        // The above call won't complete if the isolate is paused.
+        final state = await Future.any<HttpTimelineLoggingState>([
```

Comment:
Can you introduce a timeout(Future operation, int timeoutMilliseconds) to wrap this functionality?